### ISSUE TITLE
Default profile is :dev

### ISSUE THREAD

**Issue by jeaye** [taylorwood/lein-native-image#1]
Please consider using `:uberjar` as the default `lein native-image` profile. I would imagine that most people making native images are doing so with their fully AOT-compiled, production build. Currently, that requires `lein with-profile uberjar native-image`.

Thanks for the great plugin.

**Comment by taylorwood** [comment]
I believe this is a regression from the previous release which built the native image from the project's uberjar, so that profile was being used automatically—now the native image is being built from the class files. (I'm considering adding a flag for building from uberjar vs. class files.)

I've added support for a task-specific `:native-image` profile in https://github.com/taylorwood/lein-native-image/commit/7ac4bda4c43b7b6b612df714607e48f4476d7b55, and if that doesn't exist it merges the `:uberjar` profile. I deployed 0.3.0-SNAPSHOT to clojars. Let me know what you think and I'll release this week.

Thanks for reporting the issue!

**Comment by jeaye** [comment]
Thanks for the quick reply. I think this has introduced, or exposed, another issue. With the current `0.2.0` release, I can use the following lein profile to build:

```clojure
:profiles {:uberjar {:aot :all
                    :jvm-opts ["-Dclojure.compiler.direct-linking=true"]
                    :native-image {:name "easy-pack"
                                   :opts ["--verbose"
                                          "--no-server"
                                          "-H:+ReportUnsupportedElementsAtRuntime"]
                                   :graal-bin "/usr/lib/jvm/java-8-graal/bin"}}}
```

I can build it using the following:

```bash
$ lein clean; lein with-profile uberjar native-image
```

However, with the new `0.3.0-SNAPSHOT`, the same config fails to use the `uberjar` profile when I build it like this:

```bash
$ lein clean; lein native-image
Reflection warning, clojure/tools/nrepl/middleware/load_file.clj:87:38 - reference to field recv on java.lang.Object can't be resolved.
Reflection warning, clojure/tools/nrepl/middleware/load_file.clj:88:46 - call to method recv on java.lang.Object can't be resolved (no such method).
Reflection warning, clojure/tools/nrepl/middleware/load_file.clj:93:27 - call to method send on java.lang.Object can't be resolved (no such method).
Reflection warning, clojure/tools/nrepl/server.clj:142:21 - call to java.net.InetSocketAddress ctor can't be resolved.
Compiling com.jeaye.easy-pack.output
Reflection warning, me/raynes/fs.clj:517:42 - reference to field getName can't be resolved.
Compiling com.jeaye.easy-pack.util
Compiling com.jeaye.easy-pack.input
Compiling com.jeaye.easy-pack.output.css
Compiling com.jeaye.easy-pack.output.image
Compiling com.jeaye.easy-pack.cli
Compiling com.jeaye.easy-pack.layout
Compiling com.jeaye.easy-pack
Compiling com.jeaye.easy-pack.test.output.image
Reflection warning, com/jeaye/easy_pack/test.clj:31:5 - reference to field getPath can't be resolved.
Compiling com.jeaye.easy-pack.test.cli
Compiling com.jeaye.easy-pack.test
Compiling com.jeaye.easy-pack.output
Compiling com.jeaye.easy-pack.util
Compiling com.jeaye.easy-pack.input
Compiling com.jeaye.easy-pack.output.css
Compiling com.jeaye.easy-pack.output.image
Compiling com.jeaye.easy-pack.cli
Compiling com.jeaye.easy-pack.layout
Compiling com.jeaye.easy-pack
Executing [
/usr/lib/jvm/java-8-graal/bin/java \
-Xbootclasspath/a:/usr/lib/jvm/java-8-graal/jre/lib/boot/graal-sdk.jar:/usr/lib/jvm/java-8-graal/jre/lib/boot/graaljs-scriptengine.jar \
-cp \
/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/objectfile.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/svm.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/pointsto.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/jvmci-hotspot.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/jvmci-api.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/graal.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/graal-management.jar \
-server \
-d64 \
-noverify \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-XX:-UseJVMCIClassLoader \
-XX:+UseJVMCICompiler \
-Dgraal.CompileGraalWithC1Only=false \
-XX:CICompilerCount=4 \
-Dgraal.VerifyGraalGraphs=false \
-Dgraal.VerifyGraalGraphEdges=false \
-Dgraal.VerifyGraalPhasesSize=false \
-Dgraal.VerifyPhases=false \
-Dgraal.EagerSnippets=true \
-Xss10m \
-Xms1g \
-Xmx13411631104 \
-Duser.country=US \
-Duser.language=en \
-Dsubstratevm.version=b76cd1450aa1085c6b82d3d7f1563fa0c26d6b03:substratevm \
-Dgraalvm.version=1.0.0-rc2 \
-Dorg.graalvm.version=1.0.0-rc2 \
-Dcom.oracle.graalvm.isaot=true \
-Djvmci.class.path.append=/usr/lib/jvm/java-8-graal/jre/lib/jvmci/graal.jar \
com.oracle.svm.hosted.NativeImageGeneratorRunner \
-imagecp \
/usr/lib/jvm/java-8-graal/jre/lib/boot/graal-sdk.jar:/usr/lib/jvm/java-8-graal/jre/lib/boot/graaljs-scriptengine.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/objectfile.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/svm.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/builder/pointsto.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/jvmci-hotspot.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/jvmci-api.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/graal.jar:/usr/lib/jvm/java-8-graal/jre/lib/jvmci/graal-management.jar:/usr/lib/jvm/java-8-graal/jre/lib/svm/library-support.jar:/home/jeaye/projects/easy-pack/test:/home/jeaye/projects/easy-pack/src:/home/jeaye/projects/easy-pack/dev-resources:/home/jeaye/.m2/repository/fipp/fipp/0.6.6/fipp-0.6.6.jar:/home/jeaye/.m2/repository/criterium/criterium/0.4.4/criterium-0.4.4.jar:/home/jeaye/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar:/home/jeaye/.m2/repository/grimradical/clj-semver/0.3.0/clj-semver-0.3.0.jar:/home/jeaye/.m2/repository/org/clojure/data.codec/0.1.0/data.codec-0.1.0.jar:/home/jeaye/.m2/repository/slamhound/slamhound/1.5.5/slamhound-1.5.5.jar:/home/jeaye/.m2/repository/org/clojure/tools.nrepl/0.2.12/tools.nrepl-0.2.12.jar:/home/jeaye/.m2/repository/org/clojars/brenton/google-diff-match-patch/0.1/google-diff-match-patch-0.1.jar:/home/jeaye/.m2/repository/venantius/glow/0.1.4/glow-0.1.4.jar:/home/jeaye/.m2/repository/net/mikera/cljunit/0.6.0/cljunit-0.6.0.jar:/home/jeaye/.m2/repository/clojure-complete/clojure-complete/0.2.4/clojure-complete-0.2.4.jar:/home/jeaye/.m2/repository/robert/hooke/1.3.0/hooke-1.3.0.jar:/home/jeaye/.m2/repository/cider/cider-nrepl/0.15.1/cider-nrepl-0.15.1.jar:/home/jeaye/.m2/repository/org/antlr/antlr4/4.5.3/antlr4-4.5.3.jar:/home/jeaye/.m2/repository/medley/medley/1.0.0/medley-1.0.0.jar:/home/jeaye/.m2/repository/net/mikera/mathz/0.3.0/mathz-0.3.0.jar:/home/jeaye/.m2/repository/io/aviso/pretty/0.1.30/pretty-0.1.30.jar:/home/jeaye/.m2/repository/org/clojure/clojure/1.9.0/clojure-1.9.0.jar:/home/jeaye/.m2/repository/junit/junit/4.12/junit-4.12.jar:/home/jeaye/.m2/repository/mvxcvi/whidbey/1.3.0/whidbey-1.3.0.jar:/home/jeaye/.m2/repository/venantius/ultra/0.5.2/ultra-0.5.2.jar:/home/jeaye/.m2/repository/bultitude/bultitude/0.2.8/bultitude-0.2.8.jar:/home/jeaye/.m2/repository/net/mikera/mikera-gui/0.3.1/mikera-gui-0.3.1.jar:/home/jeaye/.m2/repository/instaparse/instaparse/1.4.1/instaparse-1.4.1.jar:/home/jeaye/.m2/repository/org/imgscalr/imgscalr-lib/4.2/imgscalr-lib-4.2.jar:/home/jeaye/.m2/repository/mvxcvi/puget/1.0.1/puget-1.0.1.jar:/home/jeaye/.m2/repository/org/tukaani/xz/1.5/xz-1.5.jar:/home/jeaye/.m2/repository/org/clojure/core.specs.alpha/0.1.24/core.specs.alpha-0.1.24.jar:/home/jeaye/.m2/repository/me/raynes/fs/1.4.6/fs-1.4.6.jar:/home/jeaye/.m2/repository/mvxcvi/arrangement/1.1.0/arrangement-1.1.0.jar:/home/jeaye/.m2/repository/orchestra/orchestra/2017.11.12-1/orchestra-2017.11.12-1.jar:/home/jeaye/.m2/repository/org/clojure/tools.cli/0.3.7/tools.cli-0.3.7.jar:/home/jeaye/.m2/repository/org/apache/commons/commons-compress/1.8/commons-compress-1.8.jar:/home/jeaye/.m2/repository/net/mikera/clojure-utils/0.8.0/clojure-utils-0.8.0.jar:/home/jeaye/.m2/repository/org/tcrawley/dynapath/0.2.5/dynapath-0.2.5.jar:/home/jeaye/.m2/repository/org/clojure/core.rrb-vector/0.0.11/core.rrb-vector-0.0.11.jar:/home/jeaye/.m2/repository/imagez/imagez/1.6.1/imagez-1.6.1.jar:/home/jeaye/.m2/repository/com/jhlabs/filters/2.0.235-1/filters-2.0.235-1.jar:/home/jeaye/.m2/repository/net/mikera/randomz/0.3.0/randomz-0.3.0.jar:/home/jeaye/.m2/repository/clj-antlr/clj-antlr/0.2.3/clj-antlr-0.2.3.jar:/home/jeaye/.m2/repository/org/clojure/spec.alpha/0.1.143/spec.alpha-0.1.143.jar:/home/jeaye/.m2/repository/org/antlr/antlr4-runtime/4.5.3/antlr4-runtime-4.5.3.jar \
-watchpid \
10152 \
-H:Path=/home/jeaye/projects/easy-pack \
-H:+ReportUnsupportedElementsAtRuntime \
-H:Name=/home/jeaye/projects/easy-pack/target/default+c6e091ff/easy-pack \
-H:CLibraryPath=/usr/lib/jvm/java-8-graal/jre/lib/svm/clibraries/linux-amd64 \
-H:Class=com.jeaye.easy_pack
]
   classlist:   1,408.12 ms
error: Main entry point class 'com.jeaye.easy_pack' not found.
Error: Image building with exit status 1
Failed to create native image
```

**Comment by taylorwood** [comment]
@jeaye yep, those changes created a new issue b/c I wasn't using the merged project config for the compilation step. It worked fine w/my test projects because they weren't using parameterized target paths i.e. `"target/%s"`. I was able to reproduce/fix in https://github.com/taylorwood/lein-native-image/commit/534df5484bb8911d4c05cd8f21a5eb674f6b914c and I just pushed 0.3.0-SNAPSHOT again. Thanks for your patience.
